Eevee.Sleep.Bot/Modules/SlashCommands/RoleManagementSlashModule.cs

```diff
@@ -0,0 +1,131 @@
+using Discord;
+using Discord.Interactions;
+using JetBrains.Annotations;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Utils;
+using Eevee.Sleep.Bot.Enums;
+using Discord.WebSocket;
+
+namespace Eevee.Sleep.Bot.Modules.SlashCommands;
+
+[Group("role", "Commands for managing the roles on a user.")]
+public class RoleManagementSlashModule : InteractionModuleBase<SocketInteractionContext> {
+    [SlashCommand("display", "Select which of the role you have to display.")]
+    [UsedImplicitly]
+    public async Task DisplayRoleAsync() {
+        await Context.Interaction.RespondAsync(
+            "Select your role to display.",
+            components: DiscordMessageMaker.MakeRoleSelectButton(
+                ownRoles: DiscordRoleContoller.FindRoleIdsByUserId(Context.User.Id), 
+                trackedRoles: DiscordRoleContoller.FindAllTrackedRoles(),
+                buttonId: ButtonId.RoleChanger
+            ),
+            ephemeral: true
+        );
+
+        // Ephemeral messages cannot be deleted by the button handler, so the message containing the button should be deleted after 30 seconds.
+        await Task.Delay(TimeSpan.FromSeconds(30));
```

Comment:
Refactor this into a method maybe?

Reply:
Not a generic function, so I confined it to the private method.
If you want to reuse it, you will need to refactor it to match your signature.

[54f6777](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/pull/25/commits/54f6777169a3dc207f00335140b0b49b60b25c37)